src/backend/codegen/include/codegen/utils/codegen_utils.h

```diff
@@ -249,8 +249,14 @@ class CodeGenerator {
         module_.get());
   }
 
+  template <typename FuncType>
+  llvm::Function* CreateFunctionTypeDef(
```

Comment:
1. Missing comments
2. Maybe use Typedef since that's one word in C land 
3. Why do we need this - are we not able to infer the function type automatically based on the return type?